src/components/figure/demo/basic.twig

```diff
@@ -0,0 +1,7 @@
+{% embed '../figure.twig' %}
+  {% if hasCaption %}
```

Comment:
Since you're already checking for the existence of the caption block in your template, I'm not sure why we've also added a `hasCaption` boolean here?